### PR TITLE
Use extras_require

### PR DESCRIPTION
**Pull request by cfournie**
For consistency, let's move all dependencies into `setup.py`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,8 +8,7 @@ clean:
 		rm -fr *.egg-info
 
 install:
-		pip install -e .
-		pip install -r dev_requirements.txt
+		pip install -e .[test]
```
Comment (kmtaylor-github) [inline Makefile]: I assume this is `pip` syntax? Would it be safer to quote it (to prevent free regex globbing).

Comment (honkfestival) [inline Makefile]: +1 for quoting

File: setup.py [hunk 2]
```diff
@@ -32,11 +32,26 @@
...
+            'flake8',
+            'mock',
+            'pylint',
+            'pytest-cov',
+            'pytest-randomly',
+            'pytest>=2.7',
+            'requests-mock',
+            'xmltodict',
+        ],
+    },
```
Comment (honkfestival) [inline setup.py]: Neat, TIL

File: Makefile (hunk 1 above)
Comment (cfournie) [inline Makefile]: Those are [requirement specifiers](http://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-dependencies), whose syntax, in this case, is described in [PEP 508](https://www.python.org/dev/peps/pep-0508/) as  `extras` and are comma-delimited ids within square brackets which pip interprets case-insensitively and maps to entries in `setup.py`'s `extras_require`. Quotes would be superfluous.

Comment (kmtaylor-github) [inline Makefile]: > Quotes would be superfluous.

Are you sure? Makefile lines are executed via a `sh` compatible shell. 

```
$ echo [test]
[test]
$ touch t
$ echo [test]
t
```

Comment (cfournie) [inline Makefile]: Ahh, yes, we should quote that entire pip argument. I thought that you meant that we should quote `pip install -e .["test"]`.

File: Makefile [hunk 3]
```diff
@@ -8,8 +8,7 @@ clean:
 		rm -fr *.egg-info
 
 install:
-		pip install -e .
-		pip install -r dev_requirements.txt
+		pip install -e ".[test]"
```
Comment (kmtaylor-github) [inline Makefile]: `:+1:`

**Pull request merged**